JetMETCorrections/Type1MET/plugins/MultShiftMETcorrInputProducer.h

```diff
@@ -0,0 +1,61 @@
+#ifndef JetMETCorrections_Type1MET_MultShiftMETcorrInputProducer_h
+#define JetMETCorrections_Type1MET_MultShiftMETcorrInputProducer_h
+
+/** \class MultShiftMETcorrInputProducer
+ *
+ * Compute MET correction to compensate systematic shift of MET in x/y-direction
+ * (cf. https://indico.cern.ch/getFile.py/access?contribId=1&resId=0&materialId=slides&confId=174318 )
+ *
+ * \authors Robert Schoefbeck, Vienna
+ *
+ *
+ *
+ */
+
+#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidate.h"
+
+#include "DataFormats/METReco/interface/MET.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidateFwd.h"
+
+#include <TF1.h>
+
+#include <string>
+#include <vector>
+
+class MultShiftMETcorrInputProducer : public edm::EDProducer  
```

Comment:
this needs to be made thread aware.
@Dr15Jones is TF1 thread-safe?
If it is, ```edm::stream::EDProducer<> ``` instead of the legacy edm::EDProducer should be used here